src/vcf/assembly_checker.cpp

```diff
@@ -150,12 +150,17 @@ namespace ebi
                       continue;
                   }
                   contig_name = found_synonyms[0];
-              } else if (!fetch_contig_on_demand) {
-                  if (fasta->count(contig_name) == 0) {
-                      report_missing_chromosome(line_num, record_core, outputs);
-                      is_valid = false;
-                      continue;
-                  }
+              }
+
+              if (fetch_contig_on_demand && downloadedContigs.find(contig_name)==downloadedContigs.cend()) {
```

Comment:
Add whitespaces around the comparison operator. Applies to other conditions below.